gnd/src/main/java/com/google/android/gnd/persistence/local/room/RecordMutationEntity.java

```diff
@@ -39,13 +39,25 @@
 @AutoValue
 @Entity(
     tableName = "record_mutation",
-    foreignKeys =
-        @ForeignKey(
-            entity = RecordEntity.class,
-            parentColumns = "id",
-            childColumns = "record_id",
-            onDelete = CASCADE),
-    indices = {@Index("record_id")})
+    foreignKeys = {
+      @ForeignKey(
+          entity = FeatureEntity.class,
+          parentColumns = "id",
+          childColumns = "feature_id",
+          onDelete = CASCADE),
+      @ForeignKey(
+          entity = RecordEntity.class,
+          parentColumns = "id",
+          childColumns = "record_id",
+          onDelete = CASCADE)
+    },
+    indices = {
```

Comment:
Are we going to need all these indices!? 0.o

(a facetious comment)

Reply:
No. Good question actually.

PK contraints are implemented as indexes in Sqlite, so @Index isn't needed.

Also, we shouldn't add stuff until we use it (YAGNI), and composite indexes should be used for fields that are queried together.

Updated all the things.